tutorials/plot_decoding_csp_timefreq.py

```diff
@@ -57,18 +59,56 @@
 
 # Assemble list of frequency range tuples
 freqs = np.linspace(min_freq, max_freq, n_freqs)  # assemble frequencies
-freq_ranges = zip(freqs[:-1], freqs[1:])  # make freqs into a list of tuples
+freq_ranges = list(zip(freqs[:-1], freqs[1:]))  # make freqs list of tuples
 
 # Infer window spacing from the max freq and number of cycles to avoid gaps
 window_spacing = (n_cycles / np.max(freqs) / 2.)
 centered_w_times = np.arange(tmin, tmax, window_spacing)[1:]
 n_windows = len(centered_w_times)
 
 ###############################################################################
+# Loop through frequencies, apply classifier and save scores
+
+# init scores
+freq_scores = np.zeros((n_freqs - 1,))
+
+# Loop through each frequency range of interest
+for freq, (fmin, fmax) in enumerate(freq_ranges):
+
+    # Infer window size based on the frequency being used
+    w_size = n_cycles / ((fmax + fmin) / 2.)  # in seconds
+
+    # Apply band-pass filter to isolate the specified frequencies
+    raw_filter = raw.copy().filter(fmin, fmax, n_jobs=1)
+
+    # Extract epochs from filtered data, padded by window size
+    epochs = Epochs(raw_filter, events, event_id, tmin - w_size, tmax + w_size,
+                    proj=False, baseline=None, preload=True)
+    epochs.drop_bad()
+    y = epochs.events[:, 2] - 2
```

Comment:
I don't think that it changes anything to subtract 2, does it?

Else you can use sklearn's LabelEncoder to make it explicit